developer-docs/kerberos.md

```diff
@@ -0,0 +1,73 @@
+
+The [gssapi](https://github.com/zenchild/gssapi) gem that Bolt relies on for [Generic Security Services](https://en.wikipedia.org/wiki/Generic_Security_Services_Application_Program_Interface) defaults to MIT Kerberos, but also supports Heimdal via a programmatic opt-in by adding `require 'gssapi/heimdal'` prior to `require 'gssapi'` (for instance, to the code in `winrm/connection.rb`). Unfortunately the support for Heimdal is not well tested, the paths to the Heimdal libraries are currently hard-coded, and the library doesn't account for semantic differences in the Heimdal and MIT implementations of Kerberos APIs, which can cause segfaults.
+
+Samba also has optional experimental support for compiling against MIT Kerberos instead of the default Heimdal.
+
+#### MIT Kerberos
+
+[MIT Kerberos](https://web.mit.edu/kerberos/) is the default implementation used in Linux, and there is some general information on the [client installation instructions page](https://web.mit.edu/Kerberos/www/krb5-latest/doc/admin/install_clients.html).
+
+##### RHEL
+
+RHEL typically installs with the following packages
+
+> yum install krb5-workstation krb5-libs krb5-auth-dialog
+
+RedHat maintains docs about [Configuring a Kerberos 5 client](https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/6/html/managing_smart_cards/configuring_a_kerberos_5_client)
+
+##### Ubuntu
+
+Ubuntu typically installs with the following packages
+
+> apt-get install krb5-user libpam-krb5 libpam-ccreds auth-client-config
+
+Canonical maintains docs about running a [Kerberos Linux Client](https://help.ubuntu.com/lts/serverguide/kerberos.html#kerberos-linux-client)
+
+#### Microsoft Kerberos
+
+Microsoft Kerberos is built-in to Windows and most easily consumed transparently by domain joining computers to Active Directory.
+
+Microsoft provides some guidance around [SSPI/Kerberos Interopability with GSSAPI](https://docs.microsoft.com/en-us/windows/win32/secauthn/sspi-kerberos-interoperability-with-gssapi) that covers how to map gssapi calls to their equivalent Windows APIs.
+
+### Kerberos Tooling
+
+On Linux and OSX, regardless of the package in use, client configuration is typically stored in `/etc/krb5.conf` (overridable with the `KRB5_CONFIG` environment variable). Windows Active Directory does not use such a config file.
+
+Important client tools include:
+
+* `kinit` for acquiring tickets (not used on Windows)
+* `klist` for listing tickets
+* `kdestroy` for destroying tickets (not used on Windows)
+
+### Usage
+
+Manual verification of Bolt can be performed from a Linux node that is domain joined to Active Directory using the following steps:
+
+- `kinit -C Administrator@domain.com` to acquire a TGT (ticket granting ticket)
+- `bolt command run 'whoami' --targets winrm://dc.domain.com --realm DOMAIN.COM --no-ssl-verify` to connect over HTTPS
```

Comment:
Is `--no-ssl-verify` necessary? If not I'd remove it here.

Reply:
It's likely this could be needed on Linux, but shouldn't be needed when the client is Windows.

I may note that it's optional based on environment.